src/main/scala/com/fulcrumgenomics/util/ReadStructure.scala

```diff
@@ -150,24 +141,64 @@ object ReadSegment {
     }
   }
 
-
   /** Creates a new read segment of the same type but with a new length */
   def apply(segment: ReadSegment, length: Int): ReadSegment = segment match {
     case seg: Template         => Template(seg.offset, length)
     case seg: SampleBarcode    => SampleBarcode(seg.offset, length)
     case seg: MolecularBarcode => MolecularBarcode(seg.offset, length)
     case seg: Skip             => Skip(seg.offset, length)
   }
+
+  def structureRead(segments: Seq[ReadSegment], bases: String, useOffset: Boolean = false): String = {
```

Comment:
Consider a read structure of "10B50T10B".  A read will be 70bp, but the sample barcode bases to match against in the sample sheet will be 20bp (missing the template bases obviously).  So the input to this method will be a string of length 20 (sample barcode bases read) but the three read segments ("10B", "50T", "10B") have offsets relative to a 70bp read.